### PR TITLE
Fix back button listener in Toolbar

### PR DESCRIPTION
**Pull request by janziemba**
Currently it makes back button unresponsive if you open the search bar a close it.

### CODE REVIEW

File: src/Toolbar/Toolbar.react.js [hunk 1]
```diff
@@ -157,14 +157,8 @@ const contextTypes = {
     uiTheme: PropTypes.object.isRequired,
 };
 
-const EMPTY_BACK_BUTTON_LISTENER = { remove: () => {} };
 const getBackButtonListener = (callback, isSearchActive) => {
```
Comment (Nemikolh) [inline src/Toolbar/Toolbar.react.js]: You need to remove `isSearchActive`

File: src/Toolbar/Toolbar.react.js [hunk 2]
```diff
@@ -157,14 +157,8 @@ const contextTypes = {
...
-    // if search is active by default we need to listen back button
-    if (isSearchActive) {
-        return BackAndroid.addEventListener('closeRequested', callback);
-    }
-
-    return EMPTY_BACK_BUTTON_LISTENER;
+    return BackAndroid.addEventListener('hardwareBackPress', callback);
```
Comment (Nemikolh) [inline src/Toolbar/Toolbar.react.js]: If I understand the lint error correctly this should just be:
```js
const getBackButtonListener = (callback) => 
    BackAndroid.addEventListener('hardwareBackPress', callback)
;
```

File: src/Toolbar/Toolbar.react.js [hunk 3]
```diff
@@ -185,10 +179,7 @@ class Toolbar extends PureComponent {
         super(props);
 
         const isSearchActive = props.isSearchActive || false;
-        this.backButtonListener = getBackButtonListener(
-            this.onSearchCloseRequested,
-            isSearchActive,
-        );
+        this.backButtonListener = isSearchActive ? getBackButtonListener(this.onSearchCloseRequested) : null;
```
Comment (Nemikolh) [inline src/Toolbar/Toolbar.react.js]: This line is too long (limit is 100)

**Comment by janziemba** [comment]
@Nemikolh Thank you for your comments. I have updated the code.
Now I don't understand why Codecov reports coverage decrease. 😞

**Comment by Nemikolh** [comment]
You're welcome! :)

Don't worry too much, It seems to be caused by the additional branch as you can [see here](https://codecov.io/gh/xotahal/react-native-material-ui/pull/197/src/src/Toolbar/Toolbar.react.js#L290) (line 291).

As there is currently no tests making sure that the property `searchable` is working correctly, the `onSearchCloseRequested` function is never called. 

I guess we should ask @xotahal if he wants a test for it.

**Comment by sstur** [comment]
@janziemba: It seems you are adding an event listener for hardwareBackPress but never removing it again. This breaks the main App's back handling in many cases. I'd suggest add a listener on mount and remove that listener on unmount, then inside the listener you can decide if you wish to return true (handled) or false (let another handler run).

**Comment by sstur** [comment]
I need to fork this repo to make my app work properly. I'll make a PR for the fix if I can.

**Pull request merged**